### DISCUSSION TITLE
Modo LAIR (Leia-Avalie-Imprima-Repita, ou REPL) para Potigol

### DISCUSSION THREAD

**Discussion by leonelsanchesdasilva** [potigol/potigol#64] [Ideas]
Como código em Potigol não requer uma estrutura de início e fim de um fonte, como os dialetos de Portugol, podemos imaginar um REPL (uma execução contínua por linha de comando). Há o interesse da comunidade de implementar isso também no projeto de Potigol dessa organização? Se sim, eu gostaria de contribuir.

**Comment by lrlucena** [comment]
Olá @leonelsanchesdasilva ,

Gosto bastante da ideia de criar um REPL para Potigol! Acho que seria bom para o projeto e poderia ajudar as pessoas a entenderem melhor a linguagem.

No entanto, devo confessar que não tenho experiência com a implantação de um REPL. Pensei em usar o REPL da linguagem Scala como modelo inicial, já que Potigol é implementado nela. Você tem alguma sugestão?

**Reply by leonelsanchesdasilva** [reply]
Oi, professor 👋 

Acho o REPL de Scala um bom começo. Fiz alguns experimentos nele, aliás:

![image](https://github.com/potigol/potigol/assets/53848829/e350ee4d-0308-413c-bd2d-ae91bf02d2c1)

REPLs normalmente respondem bem como calculadoras e depois como executores de código incremental. Isso significa que, enquanto o REPL está ativo, temos a manutenção de estados de variáveis e declarações de classes e funções. 

![image](https://github.com/potigol/potigol/assets/53848829/648bf55a-645b-4990-b871-d6af75dc0ee5)

Acho que a discussão pode ser usada para definir cada tipo de resposta, e comportamentos iniciais. Depois de uma primeira implementação, o comportamento pode ser amadurecido.
